### PR TITLE
call cat.claw() in flip.kick()

### PR DESCRIPTION
**Pull request by godsflaw**
Adapts the flipper for `LIQ-1.2`.

### CODE REVIEW

File: src/RwaFlipper.sol [hunk 1]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
+        uint lot,
+        uint bid
+    ) public auth returns (uint id) {
         require(kicks < uint(-1), "RwaFlipper/overflow");
         id = ++kicks;
 
-        usr; gal; tab; bid;
+        usr; gal; bid;
         vat.flux(ilk, msg.sender, address(this), lot);
+        cat.claw(tab);
```
Comment (hexonaut) [inline src/RwaFlipper.sol:52]: Should emit a `Kick(id, lot, bid, tab, usr, gal)` event.

File: src/RwaFlipper.sol [hunk 2]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
-    constructor(address vat_, bytes32 ilk_) public {
+    constructor(address vat_, address cat_, bytes32 ilk_) public {
         vat = VatLike(vat_);
+        cat = CatLike(cat_);
         ilk = ilk_;
         wards[msg.sender] = 1;
     }
 
-    function kick(address usr, address gal, uint tab, uint lot, uint bid) public auth returns (uint id) {
+    function file(bytes32 what, address data) external note auth {
```
Comment (hexonaut) [inline src/RwaFlipper.sol:35]: Remove `note`. No longer being used.

File: src/RwaFlipper.sol [hunk 3]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
+        if (what == "cat") cat = CatLike(data);
+        else revert("RwaFlipper/file-unrecognized-param");
```
Comment (hexonaut) [inline src/RwaFlipper.sol:37]: Add a `File` event.

File: src/RwaFlipper.sol [hunk 4]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
     uint256 public kicks = 0;
 
 
     // --- init ---
-    constructor(address vat_, bytes32 ilk_) public {
+    constructor(address vat_, address cat_, bytes32 ilk_) public {
         vat = VatLike(vat_);
+        cat = CatLike(cat_);
         ilk = ilk_;
         wards[msg.sender] = 1;
```
Comment (hexonaut) [inline src/RwaFlipper.sol:32]: Add `Rely` event. This should be in place for `rely` and `deny` as well.

File: src/RwaFlipper.sol [hunk 5]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
+        if (what == "cat") cat = CatLike(data);
+        else revert("RwaFlipper/file-unrecognized-param");
+    }
+
+    function kick(
+        address usr,
+        address gal,
+        uint tab,
+        uint lot,
+        uint bid
```
Comment (hexonaut) [inline src/RwaFlipper.sol:45]: Use `uint256` instead of `uint`.

File: src/RwaFlipper.sol [hunk 6]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
     }
 
     VatLike public vat;
```
Comment (hexonaut) [inline src/RwaFlipper.sol:21]: Should be immutable to save gas.

File: src/RwaFlipper.sol [hunk 7]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
+    CatLike public cat;
     bytes32 public ilk;
```
Comment (hexonaut) [inline src/RwaFlipper.sol:23]: Also should be immutable to save gas.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Some requested changes, but the overall structure looks good.

File: src/RwaFlipper.sol [hunk 8]
```diff
@@ -16,22 +19,36 @@ contract RwaFlipper {
...
+        else revert("RwaFlipper/file-unrecognized-param");
+    }
+
+    function kick(
+        address usr,
+        address gal,
+        uint tab,
+        uint lot,
+        uint bid
+    ) public auth returns (uint id) {
```
Comment (kmbarry1) [inline src/RwaFlipper.sol:46]: Can be `external` instead of `public`, I believe.

**Comment by kmbarry1** [comment]
Tests?

File: src/RwaFlipper.sol (hunk 6 above)
Comment (brianmcmichael) [reply src/RwaFlipper.sol:21]: This is currently using 0.5.12, so `immutable` isn't available here.

File: src/RwaFlipper.sol (hunk 7 above)
Comment (brianmcmichael) [reply src/RwaFlipper.sol:23]: Can't make this one immutable for reasons described above, but it's also File-able, so needs to remain editable.

Comment (hexonaut) [reply src/RwaFlipper.sol:23]: The `cat` can be filed, but not the `ilk`.

File: src/RwaFlipper.sol (hunk 6 above)
Comment (hexonaut) [reply src/RwaFlipper.sol:21]: Ah yes good point. Maybe it's worthwhile to bump up the version?

Comment (godsflaw) [reply src/RwaFlipper.sol:21]: `immutable` isn't available on this version of the compiler yet.

File: src/RwaFlipper.sol (hunk 7 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:23]: `immutable` isn't available on this version of the compiler yet.

File: src/RwaFlipper.sol (hunk 4 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:32]: This is a good point, along with all your other event points.  I think we should carry those over to the other PR.

File: src/RwaFlipper.sol (hunk 8 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:46]: I think this is just to be conformant with our other `flip.kick()` interfaces

File: src/RwaFlipper.sol (hunk 1 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:52]: Absolutely, let's carry this over to the other PR

File: src/RwaFlipper.sol (hunk 2 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:35]: I haven't broken the news to @livnev yet that we decided to move to custom events a while ago.  This is mostly to make out-of-the-box tools like etherscan more usable and to have some more custom treatment of what is indexed vs. emitted in the event.

File: src/RwaFlipper.sol (hunk 5 above)
Comment (godsflaw) [reply src/RwaFlipper.sol:45]: I also have not broken the news to @livnev that we decided to me more explicit about our `unit` bits.

**Comment by godsflaw** [comment]
OKay, going to address any issues here over in #7

**Pull request merged**